Indoor_GPS/marvelmind.py

```diff
@@ -108,14 +110,24 @@ def print_position(self):
             print ("Hedge {:d}: X: {:.3f}, Y: {:.3f}, Z: {:.3f}, Angle: {:d} at time T: {:.2f}".format(self.position()[0], self.position()[1], self.position()[2], self.position()[3], self.position()[4], self.position()[5]/1000.0))
 
     def position(self):
-        return list(self.valuesUltrasoundPosition)[-1];
+        ans = list(self.valuesUltrasoundPosition)[-1]
+        if self.num_called < 5:
```

Comment:
@SanjitBasker Does this work, remember we were debating that this function is only called when serverGUI calls it, so it would take 5 iterations of serverGUI to call this, which is quite a long time. Wouldn't we need to poll it 5 times immediately?